### PR TITLE
[SEDONA-178] Correctness issue in distance joins

### PR DESCRIPTION
**Pull request by umartin**
## Did you read the Contributor Guide?

- Yes, I have read [Contributor Rules](https://sedona.apache.org/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the assoicated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-178. The PR name follows the format `[SEDONA-178] my subject`.

## What changes were proposed in this PR?

Sedona SQL doesn't rely on Circle for distance joins. Correct results are now produces for all geometry types.

Performance could be improved but would require some refactoring of the spatial partitioning code.

See the jira for discussion.

## How was this patch tested?

All unit tests pass. New tests added.

## Did this PR include necessary documentation updates?

- Yes, I have updated the documentation update.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -68,6 +68,10 @@
         <sedona.jackson.version>2.13.3</sedona.jackson.version>
         <hadoop.version>3.2.4</hadoop.version>
         <maven.deploy.skip>false</maven.deploy.skip>
+
+        <!-- Actual scala version will be set by a profile.
+        Setting a default value helps IDE:s that can't make sense of profiles. -->
+        <scala.compat.version>2.12</scala.compat.version>
```
Comment (umartin) [inline pom.xml:74]: This is what spark does. Default values in properties section and overrides in profiles. Without it Intellij can't make sense of the project.

https://github.com/apache/spark/blob/master/pom.xml#L164

File: sql/src/test/scala/org/apache/sedona/sql/functionTestScala.scala [hunk 2]
```diff
@@ -134,14 +134,15 @@ class functionTestScala extends TestBaseScala with Matchers with GeometrySample
       var polygonDf = sparkSession.sql("select ST_GeomFromWKT(polygontable._c0) as countyshape from polygontable")
       polygonDf.createOrReplaceTempView("polygondf")
       val polygon = "POLYGON ((110.54671 55.818002, 110.54671 55.143743, 110.940494 55.143743, 110.940494 55.818002, 110.54671 55.818002))"
-      val forceXYExpect = "POLYGON ((471596.69167460164 6185916.951191288, 471107.5623640998 6110880.974228167, 496207.109151055 6110788.804712435, 496271.31937046186 6185825.60569904, 471596.69167460164 6185916.951191288))"
+      // Floats don't have an exact decimal representation. String format varies across jvm:s. Do an approximate match.
```
Comment (umartin) [inline sql/src/test/scala/org/apache/sedona/sql/functionTestScala.scala:137]: This fixes builds on java 11. The string representation of floats changed between java 8 and java 11.

There are other tests in sedona-core that fail on java 11 but this fixes sedona-sql

File: sql/src/test/scala/org/apache/sedona/sql/SpatialJoinSuite.scala [hunk 3]
```diff
@@ -128,11 +128,10 @@ class SpatialJoinSuite extends TestBaseScala with TableDrivenPropertyChecks {
       case "ST_Touches" => (l: Geometry, r: Geometry) => l.touches(r)
       case "ST_Within" => (l: Geometry, r: Geometry) => l.within(r)
       case "ST_Distance" =>
-        // XXX: ST_Distance has a weird behavior, it is wildly different from `l.distance(r)`.
```
Comment (umartin) [inline sql/src/test/scala/org/apache/sedona/sql/SpatialJoinSuite.scala:131]: This is an interesting comment!

**Comment by umartin** [comment]
It locks like r-lib/actions removed the branch "master"
https://github.com/r-lib/actions

https://github.com/r-lib/actions/issues/639

**Comment by jiayuasu** [comment]
I changed the GitHub Action to r-lib new master branch but failed at some other issues. I will merge this PR anyway as it passed all other tests.

**Review by jiayuasu — APPROVED** [review]

**Pull request merged**